presto-main/src/main/java/com/facebook/presto/connector/informationSchema/InformationSchemaPageSourceProvider.java

```diff
@@ -105,116 +100,121 @@ private InternalTable getInternalTable(ConnectorSession connectorSession, Connec
         requireNonNull(columns, "columns is null");
 
         InformationSchemaTableHandle handle = split.getTableHandle();
-        Map<String, NullableValue> filters = split.getFilters();
+        Set<QualifiedTablePrefix> prefixes = split.getPrefixes();
 
-        return getInformationSchemaTable(session, handle.getCatalogName(), handle.getSchemaTableName(), filters);
+        return getInformationSchemaTable(session, handle.getCatalogName(), handle.getSchemaTableName(), prefixes);
     }
 
-    public InternalTable getInformationSchemaTable(Session session, String catalog, SchemaTableName table, Map<String, NullableValue> filters)
+    public InternalTable getInformationSchemaTable(Session session, String catalog, SchemaTableName table, Set<QualifiedTablePrefix> prefixes)
     {
         if (table.equals(TABLE_COLUMNS)) {
-            return buildColumns(session, catalog, filters);
+            return buildColumns(session, catalog, prefixes);
         }
         if (table.equals(TABLE_TABLES)) {
-            return buildTables(session, catalog, filters);
+            return buildTables(session, catalog, prefixes);
         }
         if (table.equals(TABLE_VIEWS)) {
-            return buildViews(session, catalog, filters);
+            return buildViews(session, prefixes);
```

Comment:
Strange that `buildViews` and `buildTables` take different set of arguments.. See comment in `buildTables` how this can be changed.